src/main/java/com/javalearn/circle/data/SimpleDataReader.java

```diff
@@ -10,27 +11,32 @@
 
 public class SimpleDataReader implements DataReader {
 
+    private final static Logger LOGGER = Logger.getLogger(SimpleDataReader.class);
+
     public List<String> read (String path) throws DataException {
 
         List<String> lines = new ArrayList<String>();
         BufferedReader reader = null;
 
-        try{
-            reader = new BufferedReader(new FileReader(path));
-            String line;
-            while ((line = reader.readLine()) != null){
-                lines.add(line);
-            }
-        } catch (IOException e){
-            throw new DataException(e.getMessage(), e);
-        } finally {
-            if (reader != null){
-                try{
+        try {
+            try {
+                reader = new BufferedReader(new FileReader(path));
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    lines.add(line);
+                }
+            } catch (IOException e) {
+                LOGGER.error(e.getMessage(), e);
+                throw new DataException(e.getMessage(), e);
+
+            } finally {
+                if (reader != null) {
                     reader.close();
-                } catch (IOException e){
-                    throw new DataException(e.getMessage(), e);
                 }
             }
+        } catch (IOException e){
+            LOGGER.error(e.getMessage(), e);
+            throw new DataException(e.getMessage(), e);
```

Comment:
just log as I said before, do not throw exception from finally block since it may hide original exception if happens